eval/stack.go

```diff
@@ -9,24 +9,25 @@ import (
 
 // Returns a stack array of the current stack.
 func (s *State) Stack() []string {
-	stack := make([]string, 0, s.depth) // will be -1 because no name at toplevel but... it's fine.
-	e := s.env
-	for {
-		if e == nil {
-			break
-		}
+	if s.depth <= 1 {
+		return nil
+	}
+	stack := make([]string, 0, s.depth-1)
+	for e := s.env; e != nil; e = e.StackParent() {
```

Comment:
Can't stackparent returna nil?

Then the very next call to e.Name will panic

Reply:
> Can't stackparent return a nil?
yes it does and no it won't because that's how we stop the loop :)

Reply:
That's right

for (init, check, iterate)

Iterate is called before check right?

Reply:
iterate is called after each iteration yes, sleep deprivation is doing a number on you :)